src/components/ImageView/index.native.js

```diff
@@ -66,12 +66,18 @@ class ImageView extends PureComponent {
         if (!this.props.url) {
             return;
         }
-        ImageSize.getSize(this.props.url).then(({width, height}) => {
+        ImageSize.getSize(this.props.url).then(({width, height, rotation}) => {
             let imageWidth = width;
             let imageHeight = height;
             const containerWidth = Math.round(this.props.windowWidth);
             const containerHeight = Math.round(this.state.containerHeight);
 
+            // On specific Android devices, the dimensions are sometimes returned to us flipped here, with `rotation` set to 90 degrees.
+            // Swap them back to make sure the image fits nicely in the container. On iOS, the rotation is always undefined, so this does not apply.
+            if (rotation === 90 && imageWidth > imageHeight) {
```

Comment:
```suggestion
            if ((rotation === 90 || rotation === 270)  && imageWidth > imageHeight) {
```

Reply:
Makes sense, though does this also fix the situation from your review [here](https://github.com/Expensify/App/pull/13575#pullrequestreview-1220127578)? Because you were getting `rotation = 0`, right?

Reply:
hehe, now I see that my message was a bit confusing [here](https://github.com/Expensify/App/pull/13575#issuecomment-1354058005).

Adding 270 is not related to the case where I get `rotation = 0`, it is just for the case when I did get the rotation and the device had been rotate in 270 degrees.

I noticed that you get `rotation = 0` if you do this:

1. Open any chat
2. In the compose box, click the + button > 📎 Add attachment > 📸 Take photo
3. Take a photo in portrait mode
4. On the "Send attachment" preview page, make sure the image uses all the available width space and isn't unnecessarily squished
5. Click "Send attachment" and wait for it to upload
6. Click the attachment thumbnail: it opens the same modal the preview uses, but now `rotation = 0`.

I think this is because when our server processes the image, it looses the "rotation" information, so I was thinking that this part may need a fix in the backend

Reply:
Got it. Thanks for the clarification. Honestly I think we should just remove the `rotation` param altogether and swap the dimensions. I tested locally and it also prevents the issue you describe above with opening an image that was already uploaded. It also works with landscape images because we keep the aspect ratio of the image.